raiden_contracts/data/source/services/ServiceRegistry.sol

```diff
@@ -43,16 +43,149 @@ contract Deposit {
 
+    /// The amount of time (in seconds) till the price decreases to roughly 1/e.
+    uint256 public decay_constant = 200 days;
+
     // Once the price is at min_price, it can't decay further.
-    uint256 constant min_price = 1000;
+    uint256 public min_price;
+
+    // Whenever a deposit comes in, the price is multiplied by numerator / denominator.
+    uint256 public price_bump_numerator = 1;
+    uint256 public price_bump_denominator = 1;
+
+    // The duration of service registration/extension in seconds
+    uint256 public registration_duration = 180 days;
+
+    function change_parameters(
+            uint256 _price_bump_numerator,
+            uint256 _price_bump_denominator,
+            uint256 _decay_constant,
+            uint256 _min_price,
+            uint256 _registration_duration
+    ) public onlyController {
+        change_parameters_internal(
+            _price_bump_numerator,
+            _price_bump_denominator,
+            _decay_constant,
+            _min_price,
+            _registration_duration
+        );
+    }
+
+    function change_parameters_internal(
+            uint256 _price_bump_numerator,
+            uint256 _price_bump_denominator,
+            uint256 _decay_constant,
+            uint256 _min_price,
+            uint256 _registration_duration
+    ) internal {
+        refresh_price();
+        set_price_bump_parameters(_price_bump_numerator, _price_bump_denominator);
+        set_min_price(_min_price);
+        set_decay_constant(_decay_constant);
+        set_registration_duration(_registration_duration);
+    }
+
+    // Updates set_price to be current_price() and set_price_at to be now
+    function refresh_price() private {
+        set_price = current_price();
+        set_price_at = now;
+    }
+
+    function set_price_bump_parameters(
+            uint256 _price_bump_numerator,
+            uint256 _price_bump_denominator
+    ) private {
+        require(_price_bump_denominator > 0, "divide by zero");
+        price_bump_numerator = _price_bump_numerator;
+        price_bump_denominator = _price_bump_denominator;
+    }
+
+    function set_min_price(uint256 _min_price) private {
+        // No checks.  Even allowing zero.
+        min_price = _min_price;
+        // No checks or modifications on set_price.
+        // Even if set_price is smaller than min_price, current_price() function returns min_price.
+    }
+
+    function set_decay_constant(uint256 _decay_constant) private {
+        require(_decay_constant > 0, "attempt to set zero decay constant");
+        require(_decay_constant < 2 ** 60, "too big decay constant");
+        decay_constant = _decay_constant;
+    }
+
+    function set_registration_duration(uint256 _registration_duration) private {
+        // No checks.  Even allowing zero (when no new registrations are possible).
+        registration_duration = _registration_duration;
+    }
+
+
+    /// @notice The amount of deposits for registration or extension.
```

Comment:
"amount of deposits" sounds like you need multiple deposits, not a deposit of a specific size. Maybe "amount to deposit"?

Reply:
OK.  I also think "amount to deposit" is less ambiguous.